jvector-base/src/main/java/io/github/jbellis/jvector/util/SynchronizedGrowableBitSet.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.jbellis.jvector.util;
+
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * A thread-safe {@link BitSet} implementation that grows as needed to accommodate set(index) calls. When it
+ * does so, it will grow its internal storage multiplicatively, assuming that more growth will be
+ * needed in the future. This is the important difference from FixedBitSet + FBS.ensureCapacity,
+ * which grows the minimum necessary each time.
+ *
+ * @see GrowableBitSet
+ */
+public class SynchronizedGrowableBitSet extends BitSet {
+
+  private final java.util.BitSet bitSet;
+  private final Lock lock = new ReentrantLock();
```

Comment:
I didn't use `synchronized` blocks, because when virtual threads are used, a `synchronized` block pins virtual thread to platform (OS) thread. Locks are virtual thread friendly.

Reply:
Good thinking!